tests/suite/math/call.typ

```diff
@@ -8,6 +8,97 @@ $ pi(a,) $
 $ pi(a,b) $
 $ pi(a,b,) $
 
+--- math-call-unclosed-func ---
+#let func(x) = x
+// Error: 1-2 unclosed delimiter
+// Error: 6-7 unclosed delimiter
+// Error: 8-9 unexpected dollar sign
+$func(a$
+
+--- math-call-unclosed-non-func ---
+// Error: 1-2 unclosed delimiter
+// Error: 5-6 unclosed delimiter
+// Error: 7-8 unexpected dollar sign
+$sin(x$
+
+--- math-call-named-args ---
+#let func1(my: none) = my
+#let func2(_my: none) = _my
+#let func3(my-body: none) = my-body
+#let func4(_my-body: none) = _my-body
+#let func5(m: none) = m
+$ func1(my: a) $
+$ func2(_my: a) $
+$ func3(my-body: a) $
+$ func4(_my-body: a) $
+$ func5(m: a) $
+$ func5(m: sigma : f) $
+$ func5(m: sigma:pi) $
+
+--- math-call-named-args-duplicate ---
+#let func(my: none) = my
+// Error: 15-17 duplicate argument: my
+$ func(my: a, my: b) $
+
+--- math-call-named-single-underscore ---
+#let func(x) = x
+// Error: 8-9 expected identifier, got underscore
+$ func(_: a) $
+
+--- math-call-named-single-char-error ---
+#let func(m: none) = m
+// Error: 8-13 unexpected argument
+$ func(m : a) $
+
+--- math-call-named-args-repr ---
+#let args(..body) = body
+#let check(it, r) = test-repr(it.body.text, r)
+#check($args(_a: a)$, "(_a: [a])")
+#check($args(_a-b: a)$, "(_a-b: [a])")
+#check($args(a-b: a)$, "(a-b: [a])")
+#check($args(a-b-c: a)$, "(a-b-c: [a])")
+#check($args(a--c: a)$, "(a--c: [a])")
+#check($args(a: a-b)$, "(a: sequence([a], [−], [b]))")
+#check($args(a-b: a-b)$, "(a-b: sequence([a], [−], [b]))")
+#check($args(a-b)$, "(sequence([a], [−], [b]))")
+
+--- math-call-spread-content-error ---
+#let args(..body) = body
+// Error: 7-16 cannot spread content
+$args(..(a + b))$
+
+--- math-call-spread-multiple-exprs ---
+#let args(..body) = body
+// Error: 11-12 unexpected text
+$args(..a + b)$
+
+--- math-call-spread-unexpected-dots ---
+#let args(..body) = body
+// Error: 8-10 unexpected dots
+$args(#..range(1, 5).chunks(2))$
+
+--- math-call-spread-repr ---
+#let args(..body) = body
+#let check(it, r) = test-repr(it.body.text, r)
+#check($args(..#range(0, 4).chunks(2))$, "((0, 1), (2, 3))")
+#check($#args(range(1, 5).chunks(2))$, "(((1, 2), (3, 4)))")
+#check($#args(..range(1, 5).chunks(2))$, "((1, 2), (3, 4))")
+#check($args(#(..range(2, 6).chunks(2)))$, "(((2, 3), (4, 5)))")
+#let nums = range(0, 4).chunks(2)
+#check($args(..nums)$, "((0, 1), (2, 3))")
+#check($args(..nums;)$, "(((0, 1), (2, 3)))")
```

Comment:
In _all_ tests except for this one I can replace `..nums` with `#(0, 1), #(2, 3)` and it works correctly. This one yields a different result though. What is special about it?

Reply:
See my reply to wrzian above, I think that gives some more context and explanation here: https://github.com/typst/typst/pull/5008#discussion_r1886237086

Reply:
I did read the reply, but I still don't understand why 
```typ
#let mat(..args) = args
$ mat(#(1, 2), #(3, 4)) $
$ mat(#(1, 2), #(3, 4);) $
```
are exactly the same, but
```typ
#let mat(..args) = args
#let nums = ((1, 2), (3, 4))
$ mat(..nums) $
$ mat(..nums;) $
```
are different.

Reply:
This stumped me too for a bit, but it's just the semicolon on the code expression: `#(2, 3);`. So the parser just sees two normal arguments :upside_down_face:

Reply:
Ahhhh, so simple in the end. I was not looking forward to trying to debug this...

Reply:
oh that's so dumb. and it goes to show that maybe we should reconsider the whole semicolon termination thing.